python/lsst/dax/distribution/DataGenClient.py

```diff
@@ -0,0 +1,786 @@
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import errno
+import glob
+import os
+import re
+import shutil
+import socket
+import subprocess
+
+from DataGenConnection import DataGenConnection
+from DataIngest import DataIngest
+
+
+
+class DataGenClient:
+    """This class is used to connect to the DataGenServer and build chunks.
+
+    Parameters
+    ----------
+    host : str
+        Name of the server's host.
+    port : int
+        Server's port number.
+    target_dir : str, optional
+        The client's working directory. Clients must have different working
+        directories.
+    chunks_per_req : int, optional
+        The number of chunks wanted per request from the server.
+
+    Note
+    ----
+    This class is used to connect to the DataGenServer and uses the information
+    the server provides to generate appropriate fake chunks while reporting
+    what chunks have been created and registered with the ingest system.
+    """
+
+    def __init__(self, host, port, target_dir='fakeData', chunks_per_req=5):
+        self._host = host
+        self._port = port
+        self._target_dir = os.path.abspath(target_dir)
+        self._chunksPerReq = chunks_per_req
+        self._loop = True # set to false to end the program
+        self._gen_arg_str = None # Arguments from the server for the generator.
+        self._client = None # DataGenConnection
+        self._cfg_file_name = 'gencfg.py' # name of the local config file for the generator
+        self._cfg_file_contents = None # contents of the config file.
+        self._spec = None # spec from exec(self._cfgFileContents)
+        self._chunker = None # chunker from exec(self._cfgFileContents)
+        self._overlap = 0.017 # overlap in degrees, about 1 arcmin. TODO: This should be put
+                              # in the cfg_file from the server as changing it will change
+                              # the chunk contents.
+        self._datagenpy = '~/work/dax_data_generator/bin/datagen.py' # TODO: MUST stop hard coding this
+        self._pt_cfg_dir = 'partitionCfgs' # sub-dir of _targetDir for partitioner configs
+        self._pt_cfg_dict = None # Dictionary that stores partioner config files.
+        self.makeDir(self._target_dir)
+        self.makeDir(os.path.join(self._target_dir, self._pt_cfg_dir))
+
+        self._ingest = None
+        self._skip_ingest = True
+        self._db_name = ''
+        self._transaction_id = -1
+
+    def _setIngest(self, ingest_dict):
```

Comment:
Function should be lowercase.

Reply:
There's some discrepancy between PEP8 and the DM Python style guide when it comes to variable and function names in python. I prefer camel case, but I'll use whichever form people feel is more appropriate.